addons/mail/static/src/emoji_picker/emoji_picker.xml

```diff
@@ -17,22 +20,53 @@
             </div>
             <hr class="mx-2 my-0 text-muted"/>
         </div>
+        <t t-set="itemIndex" t-value="0"/>
         <div class="o-mail-EmojiPicker-content overflow-auto d-flex flex-wrap align-content-start align-items-center" t-ref="emoji-grid" t-on-click="selectEmoji" t-on-scroll="highlightActiveCategory">
+            <t t-if="recentEmojis.length > 0 and !state.searchStr">
+                <t t-call="mail.EmojiPicker.section">
+                    <t t-set="category" t-value="recentCategory"/>
+                </t>
+                <t t-foreach="recentEmojis" t-as="emoji" t-key="'recent_' + emoji_index">
+                    <t t-call="mail.EmojiPicker.emoji">
+                        <t t-set="emoji" t-value="emoji"/>
+                    </t>
+                    <t t-set="itemIndex" t-value="itemIndex + 1"/>
+                </t>
+            </t>
             <t t-set="current" t-value=""/>
             <t t-foreach="getEmojis()" t-as="emoji" t-key="emoji_index">
                 <t t-if="!state.searchStr and current !== emoji.category">
                     <t t-set="current" t-value="emoji.category"/>
-                    <t t-set="categorySortId" t-value="categories.find(c => c.name === current).sortId"/>
-                    <span class="w-100 fs-5 p-2 position-sticky top-0 bg-white" t-esc="current" t-att-data-category="categorySortId"/>
-                    <span class="o-mail-EmojiPicker-category opacity-100" t-att-data-category="categorySortId"/>
+                    <t t-set="category" t-value="categories.find(c => c.name === current)"/>
```

Comment:
undefined safe？

Reply:
This assumes all `emojis.category` is properly set in `emoji_data`, which was already the assumption the previous code made.

It's actually better here, because it crashes if we made a typo in `emoji_data`. And also `current` is the non-translated name, while now it displays the locale-friendly name of the category.